### PR TITLE
mint -> faucet cherry-pick 1.1

### PR DESCRIPTION
**Pull request by davidiw**
Completes the rename of mint to faucet, which is important as we try to make our documentation and code layout human accessible. This is required due to hooks in our configuration :/

no impact on diem-node itself

# Motivation
## Justification

Required to rename this due to CI/CD quirks

## Breaking Nature

None

## Who does this affect?

No one

## What happens if we don't have this?

We have to live with a poorly named folder (mint) that holds the container (faucet)

## What are the workarounds / alternatives?

Lots of work on decoupling some work on how CI/CD builds images

### CODE REVIEW

File: docker/compose/faucet/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Generated files from running docker containers
+waypoint.txt
```
Comment (sausagee) [inline docker/compose/faucet/.gitignore:2]: this looks like a commit other than https://github.com/diem/diem/pull/7610

Comment (davidiw) [reply docker/compose/faucet/.gitignore:2]: I can remove it if there's a concern. 7610 follows on top of some other cleanups in the space,

**Review by sausagee — APPROVED** [review]

**Comment by aching** [comment]
Can you please follow the formatting of the PR as in https://github.com/diem/diem/pull/7669 ?

**Pull request merged**